qiskit/transpiler/passes/optimization/optimize_1q_decomposition.py

```diff
@@ -112,13 +112,18 @@ def _substitution_checks(self, dag, old_run, new_circ, basis, qubit):
         # if we're outside of the basis set, we're obligated to logically decompose.
         # if we're outside of the set of gates for which we have physical definitions,
         #    then we _try_ to decompose, using the results if we see improvement.
+        new_error = _error(new_circ, self._target, qubit)
+        if isinstance(new_error, tuple):
+            error_rate = new_error[0]
+        else:
+            error_rate = new_error
```

Comment:
If `target is None`, the output of `_error` is the length of the circuit, which isn't really an error rate.  Is that what you meant to happen?  If so, we could change the function to just return `(len(circuit),)` instead to simplify some checking.